app/screens/more_dms/more_dms.js

```diff
@@ -208,42 +231,81 @@ class MoreDirectMessages extends PureComponent {
     };
 
     handleSelectUser = (id) => {
-        this.setState((prevState) => {
-            const wasSelected = prevState.selectedIds[id];
+        const {currentUserId} = this.props;
 
-            // Prevent selecting too many users
-            if (!wasSelected && Object.keys(prevState.selectedIds).length >= General.MAX_USERS_IN_GM - 1) {
-                return {};
-            }
+        if (id === currentUserId) {
+            const selectedId = {};
+            selectedId[currentUserId] = true;
 
-            const selectedIds = {...prevState.selectedIds};
+            this.startConversation(selectedId);
+        } else {
+            this.setState((prevState) => {
+                const {
+                    profiles,
+                    selectedCount,
+                    selectedIds
+                } = prevState;
+
+                const wasSelected = selectedIds[id];
+
+                // Prevent selecting too many users
+                if (!wasSelected && Object.keys(selectedIds).length >= General.MAX_USERS_IN_GM - 1) {
+                    return {};
+                }
 
-            if (wasSelected) {
-                Reflect.deleteProperty(selectedIds, id);
-            } else {
-                selectedIds[id] = true;
-            }
+                const newSelectedIds = Object.assign({}, selectedIds);
 
-            return {
-                selectedIds,
-                selectedCount: Object.keys(selectedIds).length
-            };
-        });
+                let newProfiles = profiles;
+                if (wasSelected) {
+                    Reflect.deleteProperty(newSelectedIds, id);
+                    if (selectedCount === 1) {
```

Comment:
i don't quite understand this

Reply:
This section handles selected user from 1 to 0. This condition resets the slice of profiles including the user's self.